core/src/main/java/org/apache/iceberg/view/ViewMetadata.java

```diff
@@ -479,21 +488,21 @@ public ViewMetadata build() {
           metadataLocation);
     }
 
-    static List<ViewVersion> expireVersions(
-        Map<Integer, ViewVersion> versionsById, int numVersionsToKeep) {
-      // version ids are assigned sequentially. keep the latest versions by ID.
-      List<Integer> ids = Lists.newArrayList(versionsById.keySet());
-      ids.sort(Comparator.reverseOrder());
+    @VisibleForTesting
+    static List<ViewVersion> expireVersions(List<ViewVersion> versions, int numVersionsToKeep) {
+      List<ViewVersion> versionsToKeep = Lists.newArrayList(versions);
 
-      List<ViewVersion> retainedVersions = Lists.newArrayList();
-      for (int idToKeep : ids.subList(0, numVersionsToKeep)) {
-        retainedVersions.add(versionsById.get(idToKeep));
+      // version ids are assigned sequentially. keep the latest numVersionsToKeep
+      if (numVersionsToKeep > 0 && versionsToKeep.size() > numVersionsToKeep) {
```

Comment:
not sure how defensive we want to be with this check but the method is package-private so someone could actually use invalid values for `numVersionsToKeep`

Reply:
If we want to be defensive since it's exposed within the package, I'd just put a precondition check to fail if it's negative.

Reply:
It's fine not to be defensive for package-private methods that are only exposed for testing. If someone uses this, then it is their responsibility to worry about validation.